compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -153,6 +155,107 @@ std::vector<std::string> split_name_input(const std::string &str)
   return ::split_into_vector(str, ',');
 }
 
+std::unique_ptr<loco::Graph> make_graph(const std::vector<const luci::CircleNode *> nodes)
+{
+  auto graph = loco::make_graph();
+
+  luci::CloneContext ctx;
+  // clone nodes
+  for (const auto &n : nodes)
+  {
+    auto clone = luci::clone_node(n, graph.get());
+    ctx.emplace(n, clone);
+  }
+  // set graph input
+  for (const auto &n : nodes)
+  {
+    for (uint32_t i = 0; i < n->arity(); i++)
+    {
+      auto arg = n->arg(i);
+      auto input_node = dynamic_cast<luci::CircleNode *>(arg);
+      auto ctx_it = ctx.find(input_node);
+      // check if the node already has been cloned
+      if (ctx_it != ctx.end())
+        continue;
+      // the node isn't graph input if it is an other node's input
+      if (std::find(nodes.begin(), nodes.end(), arg) != nodes.end())
+        continue;
+      auto circle_const = dynamic_cast<luci::CircleConst *>(arg);
+      if (circle_const == nullptr)
+      {
+        // circle input
+        auto circle_input = graph->nodes()->create<luci::CircleInput>();
+        input_node = dynamic_cast<luci::CircleNode *>(arg);
+        luci::copy_common_attributes(input_node, circle_input);
+        ctx.emplace(input_node, circle_input);
+        // graph input
+        auto graph_input = graph->inputs()->create();
+        graph_input->name(circle_input->name());
+        graph_input->dtype(circle_input->dtype());
+        // graph input shape
+        auto input_shape = std::make_unique<loco::TensorShape>();
+        input_shape->rank(circle_input->rank());
+        for (uint32_t i = 0; i < circle_input->rank(); i++)
+        {
+          if (circle_input->dim(i).known())
+          {
+            circle_input->dim(i).set(circle_input->dim(i).value());
+          }
+        }
+        graph_input->shape(std::move(input_shape));
+
+        circle_input->index(graph_input->index());
+      }
+      else
+      {
+        auto clone = luci::clone_node(circle_const, graph.get());
+        ctx.emplace(circle_const, clone);
+      }
+    }
+  }
+  // set graph output
+  for (const auto &n : nodes)
+  {
+    auto outputs = loco::succs(n);
+    for (const auto &o : outputs)
+    {
```

Comment:
I don't understand this block. Can you please explain what your intention is?

Reply:
@seanshpark This block finds the graph output like above finding graph input logic does.

The output node is not others' output.
```cxx
if (std::find(nodes.begin(), nodes.end(), o) != nodes.end())
        continue;
```

All the nodes that went through above condition will be graph outputs.